sparkle/output/configuration_output.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+"""Sparkle class to organise configuration output."""
+
+from __future__ import annotations
+
+from sparkle.CLI.help import global_variables as gv
+from sparkle.platform import \
+    generate_report_for_configuration as sgrfch
+from sparkle.platform.settings_objects import Settings
+from sparkle.solver import Solver
+from sparkle.instance import InstanceSet
+from sparkle.configurator.configuration_scenario import ConfigurationScenario
+from sparkle.configurator.configurator import Configurator
+from sparkle.solver.validator import Validator
+from sparkle.output.structures import ValidationResults, ConfigurationResults
+
+from sparkle.CLI.help.nicknames import resolve_object_name
+
+import json
+from pathlib import Path
+
+
+class ConfigurationOutput:
+    """Class that collects configuration data and outputs it a JSON format."""
+
+    def __init__(self: ConfigurationOutput, path: Path, solver: Solver,
+                 configurator: Configurator, instance_set_train: InstanceSet,
+                 instance_set_test: InstanceSet,
+                 penalty_multiplier: int, output: Path) -> None:
+        """Initialize Configurator Output class.
+
+        Args:
+            path: Path to configuration output directory
+            solver: Solver object
+            configurator: The configurator that was used
+            instance_set_train: Instance set used for training
+            instance_set_test: Instance set used for testing
+            penalty_multiplier: penalty multiplier that is applied to the par performance
+                [To be Removed]
+            output: Path to the output directory
+        """
+        self.solver = solver
+        self.configurator = configurator
+        self.instance_set_train = instance_set_train
+        self.instance_set_test = instance_set_test
+        self.penalty_multiplier = penalty_multiplier
+        self.directory = path
+
+        if output.is_dir():
+            self.output = output / "configuration.json"
+        else:
+            self.output = output
+
+        # TODO: Fix this spaghetti code to find the scenario file
+        solver_dir_name = path.name
```

Comment:
Don't forget to fix this

Reply:
I talked to Jeroen about this, if the knows a reliable way to get the scenario information if I can't just get them from the global settings due to possible changes
He told me that that is something which is currently missing and we agreed that this isn't really in the scope of the ticket since it is a bigger problem
However I wrote this down as a comment on the ticket and it should be made into a new ticket